### PR TITLE
[libc][time][windows] implement clock_getres

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
This PR implements `clock_getres` for windows.

### CODE REVIEW

**Comment by SchrodingerZhu** [comment]
should be increment. not adjustment.

File: libc/src/__support/time/windows/qpc.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//===--- Cached QPC Frequency  ----------------------------------*- C++ -*-===//
```
Comment (s-barannikov) [inline libc/src/__support/time/windows/qpc.h:1]: FWIW https://github.com/llvm/llvm-project/pull/118553

File: libc/src/__support/time/windows/qpc.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+#include "src/__support/CPP/atomic.h"
+#include "src/__support/common.h"
+
+#define WIN32_LEAN_AND_MEAN
+#define NOMINMAX
+#include <Windows.h>
+
+namespace LIBC_NAMESPACE_DECL {
+namespace qpc {
```
Comment (michaelrj-google) [inline libc/src/__support/time/windows/qpc.h:17]: `qpc` isn't an acronym I'm familiar with in this context. I'd recommend expanding it for readability.

**Review by michaelrj-google — COMMENTED** [review]
Overall LGTM

File: libc/src/__support/time/windows/qpc.h (hunk 2 above)
Comment (SchrodingerZhu) [reply libc/src/__support/time/windows/qpc.h]: fixed

File: libc/src/time/windows/clock_getres.cpp [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    libc_errno = EINVAL;
+    return -1;
+
+  case CLOCK_MONOTONIC: {
+    long long freq = performance_counter::get_ticks_per_second();
+    __builtin_assume(freq != 0);
+    // division of 1 second by frequency, rounded up.
+    long long tv_sec = static_cast<long long>(freq == 1);
+    long long tv_nsec =
+        LIBC_LIKELY(freq != 1) ? 1ll + ((1_s_ns - 1ll) / freq) : 0ll;
```
Comment (Lancern) [inline libc/src/time/windows/clock_getres.cpp:59]: Just curious: is there any reason we have to explicitly round the division results up here?

**Review by Lancern — APPROVED** [review]

**Pull request merged**